### PR TITLE
Add spatial decomposition

### PR DESCRIPTION
**Pull request by scottgigante-immunai**
## Describe your changes

Add "spatial decomposition" task from v1. This is a draft PR because I want to check I've done this part right before I move on to the next part. @rcannood can you please take a look?

## Checklist before requesting a review
- [X] I have performed a self-review of my code

- Check the correct box. Does this PR contain:
  - [ ] Breaking changes
  - [X] New functionality
  - [ ] Major changes
  - [ ] Minor changes
  - [ ] Bug fixes

- [ ] Proposed changes are described in the CHANGELOG.md

- [ ] CI Tests succeed and look good!

### CODE REVIEW

File: src/tasks/spatial_decomposition/api/anndata_output.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - type: integer
+        name: counts
+        description: Raw counts
+    obsm:
+      - type: float
+        name: coordinates
+        description: XY coordinates for each spot
+      - type: float
+        name: proportions
+        description: Esitmated cell type proportions for each spot
```
Comment (rcannood) [inline src/tasks/spatial_decomposition/api/anndata_output.yaml:17]:
```suggestion
        description: Estimated cell type proportions for each spot

**Review by rcannood — APPROVED** [review]
LGTM!

Would it make sense to create a component for generating a README based on the API files you created? If some of them are formatted wrong, the component can throw an error.

**Comment by scottgigante-immunai** [comment]
Perhaps that component could also paste the task info from task_info.yaml into the README for a more readable github repo?

**Comment by rcannood** [comment]
If you merge the main branch into this one, you should be able to generate a task readme by running viash run `src/common/create_task_readme/config.vsh.yaml -- --task spatial_decomposition` :) (Added in #182)

The `create_task_readme` doesn't really validate the task API files, but the json schema files that were added in #183 will help in making sure that the format is correct.

**Comment by rcannood** [comment]
Closing this PR since it's included in #283.

**Pull request closed without merging**